src/libraries/System.Text.Json/tests/Common/CollectionTests/CollectionTests.Memory.cs

```diff
@@ -0,0 +1,155 @@
+{
+    public abstract partial class CollectionTests
+    {
+        private static readonly byte[] s_testData = "This is some test data!!!"u8.ToArray();
+
+        [Fact]
+        public async Task SerializeMemoryOfTAsync()
+        {
+            Memory<int> memoryInt = new int[] { 1, 2, 3 }.AsMemory();
+            Assert.Equal("[1,2,3]", await Serializer.SerializeWrapper(memoryInt));
+
+            Memory<EmptyClass> memoryPoco = new EmptyClass[] { new(), new(), new() }.AsMemory();
+            Assert.Equal("[{},{},{}]", await Serializer.SerializeWrapper(memoryPoco));
+        }
+
+        [Fact]
+        public async Task SerializeReadOnlyMemoryOfTAsync()
+        {
+            ReadOnlyMemory<int> memoryInt = new int[] { 1, 2, 3 }.AsMemory();
+            Assert.Equal("[1,2,3]", await Serializer.SerializeWrapper(memoryInt));
+
+            ReadOnlyMemory<EmptyClass> memoryPoco = new EmptyClass[] { new(), new(), new() }.AsMemory();
+            Assert.Equal("[{},{},{}]", await Serializer.SerializeWrapper(memoryPoco));
+        }
+
+        [Fact]
+        public async Task DeserializeMemoryOfTAsync()
+        {
+            Memory<int> memoryInt = await Serializer.DeserializeWrapper<Memory<int>>("[1,2,3]");
+            AssertExtensions.SequenceEqual(new int[] { 1, 2, 3 }, memoryInt.Span);
+
+            Memory<EmptyClass> memoryPoco = new EmptyClass[] { new(), new(), new() }.AsMemory();
+            Assert.Equal(3, memoryPoco.Length);
+        }
+
+        [Fact]
+        public async Task DeserializeReadOnlyMemoryOfTAsync()
+        {
+            ReadOnlyMemory<int> memoryInt = await Serializer.DeserializeWrapper<ReadOnlyMemory<int>>("[1,2,3]");
+            AssertExtensions.SequenceEqual(new int[] { 1, 2, 3 }, memoryInt.Span);
+
+            ReadOnlyMemory<EmptyClass> memoryPoco = await Serializer.DeserializeWrapper<ReadOnlyMemory<EmptyClass>>("[{},{},{}]");
+            Assert.Equal(3, memoryPoco.Length);
+        }
+
+        [Fact]
+        public async Task SerializeMemoryOfTClassAsync()
+        {
+            MemoryOfTClass<int> memoryOfIntClass = new MemoryOfTClass<int>();
+            memoryOfIntClass.Memory = new int[] { 1, 2, 3 };
+
+            string json = await Serializer.SerializeWrapper(memoryOfIntClass, new JsonSerializerOptions { WriteIndented = true });
+            Assert.Equal(@"{
+  ""Memory"": [
+    1,
+    2,
+    3
+  ]
+}".NormalizeLineEndings(), json);
+        }
+
+        [Fact]
+        public async Task SerializeReadOnlyMemoryOfTClassAsync()
+        {
+            ReadOnlyMemoryOfTClass<int> memoryOfIntClass = new ReadOnlyMemoryOfTClass<int>();
+            memoryOfIntClass.ReadOnlyMemory = new int[] { 1, 2, 3 };
+
+            string json = await Serializer.SerializeWrapper(memoryOfIntClass, new JsonSerializerOptions { WriteIndented = true });
+            Assert.Equal(@"{
+  ""ReadOnlyMemory"": [
+    1,
+    2,
+    3
+  ]
+}".NormalizeLineEndings(), json);
```

Comment:
Why is indentation necessary for this test?

Reply:
I don't think is necessary, will undo it. I got inspired by these existing tests:
https://github.com/dotnet/runtime/blob/693eef6417353d45d126fa05365665bccf5b82d9/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/CustomObjectConverterTests.cs#L113-L124